Fix type errors that happens when matching up dependency versions with cypress.

When we update typescript version to `3.5.3` and `@types/ramda` to `0.25.47` which are used in cypress, there are a few type errors like these.

```
src/api.ts:449:9 - error TS2769: No overload matches this call.
  Overload 1 of 6, '(fn: (x: BindOptions) => CustomFormats | undefined, list: readonly BindOptions[]): (CustomFormats | undefined)[]', gave the following error.
    Argument of type '<T>(obj: Record<"formats", T>) => T' is not assignable to parameter of type '(x: BindOptions) => CustomFormats | undefined'.
      Types of parameters 'obj' and 'x' are incompatible.
        Type 'BindOptions' is not assignable to type 'Record<"formats", CustomFormats | undefined>'.
          Property 'formats' is optional in type 'BindOptions' but required in type 'Record<"formats", CustomFormats | undefined>'.
  Overload 2 of 6, '(fn: (x: number | BindOptions | (() => string) | { (...items: ConcatArray<BindOptions>[]): BindOptions[]; (...items: (BindOptions | ConcatArray<BindOptions>)[]): BindOptions[]; } | ... 17 more ... | (() => IterableIterator<...>)) => any, list: BindOptions[]): readonly any[]', gave the following error.
    Argument of type '<T>(obj: Record<"formats", T>) => T' is not assignable to parameter of type '(x: number | BindOptions | (() => string) | { (...items: ConcatArray<BindOptions>[]): BindOptions[]; (...items: (BindOptions | ConcatArray<BindOptions>)[]): BindOptions[]; } | ... 17 more ... | (() => IterableIterator<...>)) => any'.
      Types of parameters 'obj' and 'x' are incompatible.
        Type 'number | BindOptions | (() => string) | { (...items: ConcatArray<BindOptions>[]): BindOptions[]; (...items: (BindOptions | ConcatArray<BindOptions>)[]): BindOptions[]; } | ... 17 more ... | (() => IterableIterator<...>)' is not assignable to type 'Record<"formats", CustomFormats | undefined>'.
          Type 'number' is not assignable to type 'Record<"formats", CustomFormats | undefined>'.
  Overload 3 of 6, '(fn: (x: BindOptions) => CustomFormats | undefined, obj: Functor<BindOptions>): Functor<CustomFormats | undefined>', gave the following error.
    Argument of type '<T>(obj: Record<"formats", T>) => T' is not assignable to parameter of type '(x: BindOptions) => CustomFormats | undefined'.
      Types of parameters 'obj' and 'x' are incompatible.
        Type 'BindOptions' is not assignable to type 'Record<"formats", CustomFormats | undefined>'.

449     map(prop('formats'), options),
            ~~~~~~~~~~~~~~~


src/fill.ts:32:7 - error TS2345: Argument of type '(string | number)[]' is not assignable to parameter of type 'readonly string[]'.
  Type 'string | number' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.

32       missingProperties,
         ~~~~~~~~~~~~~~~~~

src/trim.ts:25:13 - error TS2322: Type 'PlainObject' is not assignable to type 'never'.

25             trimmedObj[prop] = trimBySchema(schema[prop], object[prop])
               ~~~~~~~~~~~~~~~~

src/trim.ts:27:13 - error TS2322: Type 'PlainObject[]' is not assignable to type 'never'.

27             trimmedObj[prop] = map(trimBySchema(schema[prop]), object[prop])

```

--- a/package.json
+++ b/package.json
@@ -47,7 +47,7 @@
   "devDependencies": {
     "@types/lodash.camelcase": "4.3.6",
     "@types/node": "9.6.55",
-    "@types/ramda": "0.25.34",
+    "@types/ramda": "0.25.47",
     "ava": "2.4.0",
     "ava-ts": "0.25.2",
     "common-tags": "1.8.0",
@@ -59,7 +59,7 @@
     "terminal-banner": "1.1.0",
     "ts-node": "7.0.1",
     "tslint": "5.20.0",
-    "typescript": "3.3.4000"
+    "typescript": "3.7.2"
   },
   "repository": {
     "type": "git",

--- a/src/api.ts
+++ b/src/api.ts
@@ -446,7 +446,7 @@ export const bind = (...options: BindOptions[]) => {
 
   const allFormats: CustomFormats[] = filter(
     exists,
-    map(prop('formats'), options),
+    map(prop('formats'), options as Required<BindOptions>[]),
   )
   const formats = mergeFormats(allFormats)
 

--- a/src/fill.ts
+++ b/src/fill.ts
@@ -16,8 +16,8 @@ export const fillBySchema = curry(
     const schemaProps = keys(schema)
 
     const missingProperties = difference(schemaProps, objectProps)
-    const filledObject = reduce(
-      (result: PlainObject, key: string): PlainObject => {
+    const filledObject = reduce<string | number, PlainObject>(
+      (result: PlainObject, key: string | number): PlainObject => {
         const property = schema[key]
         if ('defaultValue' in property) {
           const value = property.defaultValue

--- a/src/trim.ts
+++ b/src/trim.ts
@@ -18,7 +18,7 @@ export const trimBySchema = curry(
     schema = schema.properties || (schema.schema || schema.items).properties
     const objectProps = keys(object)
     const schemaProps = keys(schema)
-    return reduce(
+    return reduce<never, PlainObject>(
       (trimmedObj, prop) => {
         if (contains(prop, schemaProps)) {
           if (object[prop] && isJsonSchema(schema[prop])) {

